Test/Utils/ContrailAPI/ConfigureDNS.Tests.ps1

```diff
@@ -0,0 +1,167 @@
+
+            {
+                Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                    -DNSServerUuid $ServerUUID -Force
+            } | Should -Not -Throw
+        }
+    }
+
+    Context 'Setting DNS modes' {
+        It 'throws when wrong dns mode specified' {
+            {
+                Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                    -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                    -DNSMode 'wrongdnsmode'
+            } | Should -Throw
+        }
+
+        Context 'mode - none' {
+            It 'can set DNS mode to none' {
+                {
+                    Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                        -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                        -DNSMode 'none'
+                } | Should -Not -Throw
+            }
+        }
+
+        Context 'mode - default' {
+            It 'can set DNS mode to default' {
+                {
+                    Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                        -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                        -DNSMode 'default-dns-server'
+                } | Should -Not -Throw
+            }
+        }
+
+        Context 'mode - tenant' {
+            It 'can set DNS mode to tenant without DNS servers' {
+                {
+                    Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                        -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                        -DNSMode 'tenant-dns-server'
+                } | Should -Not -Throw
+            }
+
+            It 'can set DNS mode to tenant with DNS servers' {
+                {
+                    Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                        -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                        -DNSMode 'tenant-dns-server' -TenantServersIPAddresses @("1.1.1.1", "2.2.2.2")
+                } | Should -Not -Throw
+            }
+        }
+
+        Context 'mode - virtual' {
+            It 'can set DNS "mode" to virtual' {
+                $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                    -DNSServerName "CreatedByPS1ScriptForIpam"
+
+                {
+                    Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                        -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+                        -DNSMode 'virtual-dns-server' -VirtualServerName "CreatedByPS1ScriptForIpam"
+                } | Should -Not -Throw
+
+                Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID -Force
```

Comment:
If assert fails, this tear down step won't be called. Either use BeforeEach/AfterEach combo, or try..finally.